crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -1289,15 +1292,27 @@ async fn finalize_checkpoint(
     let checkpoint_acc =
         accumulator.accumulate_checkpoint(effects, checkpoint.sequence_number, epoch_store)?;
 
-    if let Some(path) = data_ingestion_dir {
-        store_checkpoint_locally(
-            path,
+    if data_ingestion_dir.is_some() || state.rest_index.is_some() {
+        let checkpoint_data = load_checkpoint_data(
             checkpoint,
             object_cache_reader,
             transaction_cache_reader,
             checkpoint_store,
-            tx_digests.to_vec(),
+            tx_digests,
         )?;
+
+        //TODO Do we need to ensure this happens in-order?
```

Comment:
I don't necessarily think its required to do the indexing in checkpoint order. I couldn't really think of many instances where we'd have bad races. Lets leave it where it is for now and then we can re-evaluate later.

> Just curious why not temporary hook it into existing index_tx function?

Because that function works on an individual tx basis and is on the critical path which i believe we wanted to change anyway.

Reply:
I think this is kind of worse. finalize_checkpoint is also on critical path and there's no parallelism, all checkpoints are processed one by one.
Regarding races(again there are no races here cause it's basically serialized), owner index is sensitive to concurrency and ordering (you want to process adds and removals in correct order)

Reply:
Ok, I'll update the TODO and we can revisit exactly where to discuss in a followup PR

Reply:
>Regarding races(again there are no races here cause it's basically serialized), owner index is sensitive to concurrency and ordering (you want to process adds and removals in correct order)

right, but figuring out what needs to be written to the DB and actually doing the writes could be done in separate steps so that the first part could be done in parallel and the commits to the db happen serially.